### PR TITLE
Branch is ready but many points to discuss

### PR DESCRIPTION
**Pull request by mahrossi**
I'll just open this discussion since the real expert in the whole business here is Yair, who implemented the trust radius method in this branch. Over the course of the implementation we found many issues with the way the geometry optimizers are implemented in the main geop-dev branch. All the methods work, but there are many problems related to how dependencies are passed all over the place (restart files become broken, it is hard to tell why, how, and where some quantities actually get updated, etc.). The trust radius algorithm implementation thus looks a bit different than the others. It is not a 2 line fix to fix all the other algorithms, so it is a question of how to proceed.

### CODE REVIEW

File: ipi/engine/motion/geop.py [hunk 1]
```diff
@@ -297,6 +347,84 @@ def step(self, step=None):
...
+        """ Does one simulation time step.
+            Attributes:
+            ptime: The time taken in updating the velocities.
+            qtime: The time taken in updating the positions.
+            ttime: The time taken in applying the thermostat steps.
+	    tr   : current trust radius
+        """
+
+        self.ptime = 0.0
+        self.ttime = 0.0
```
Comment (mahrossi) [inline ipi/engine/motion/geop.py]: Do you need these variables/lines here?

Comment (litman90) [inline ipi/engine/motion/geop.py]: No. Thank you for the observation.

**Comment by litman90** [comment]
I pushed an updated version with the "mapper philosophy". Could you have a look?
Still, we have the problem that we are calculating the forces for each accepted movement twice (one with the mapper and the other with the "main machinery")

**Comment by mahrossi** [comment]
Merging this branch to geop-dev as it should not break anything and is cleaner.

**Pull request merged**